source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -883,22 +888,28 @@ def fill_journeys(self, request_type, api_request, instance):
                 itertools.product(tmp2, qualified_journeys),
             )
 
-            pool1, pool2 = itertools.tee(journeys_pool)
-            journey_filter.filter_similar_vj_journeys(pool1, api_request)
-
-            if api_request['no_shared_section']:
-                journey_filter.filter_shared_sections_journeys(pool2, api_request)
+            journey_filter.filter_similar_vj_journeys(journeys_pool, api_request)
 
             responses.extend(new_resp)  # we keep the error for building the response
 
             nb_qualified_journeys = nb_journeys(responses)
 
-            # We allow one more call to kraken if there is no valid journey.
+            if nb_previously_qualified_journeys == nb_qualified_journeys:
+                # If there is no additional qualified journey in the kraken response,
+                # another request is sent to try to find more journeys, just in case...
+                if last_chance_retry:
+                    break
+                last_chance_retry = True
+            nb_previously_qualified_journeys = nb_qualified_journeys
+
             if nb_qualified_journeys == 0:
                 min_journeys_calls = max(min_journeys_calls, 2)
-
+                
         logger.debug('nb of call kraken: %i', nb_try)
 
+        if api_request['no_shared_section']:
+            journey_filter.filter_shared_sections_journeys(journey_filter.get_qualified_journeys(responses), api_request)
```

Comment:
depends if you are in debug mode

Reply:
The final filter has its purpose : to delete journeys with the same lines sequence. So this filter has nothing to do with it. I don't see the point.
Also, the journeys filtered here will be handled as every other journeys in every other filters, (mark_as_dead in the function '_filter_similar_journeys')